# Curso PCA e Clustering IEEE CIS UnB

## Aula 1 - Python Básico

Olá! Seja bem vindo ao curso de PCA e Clustering, uma parceria entre o IEEE CIS e a Estat.

Nesse curso, você vai aprender mais sobre esses dois métodos de tratamento, mas, primeiro, vamos falar um pouco sobre Python Básico.

Este notebook é dividido em seções para você poder executar uma trilha de aprendizado passo a passo no conteúdo de python básico. A ideia é que você faça ele junto com a gente na aula.

## Parte 1 - Hello Python

Dentre as coisas mais interessantes do python, a gente sempre tem que mencionar a facilidade da linguagem e seu alto potencial de escalabilidade. Para começar, vamos passar para vocês o básico da sintaxe da linguagem.

Vamos mostrar no terminal "Hello Python" de 3 formas diferentes

A primeira delas e mais simples é por meio da sintaxe padrão do python para mostrar coisas no terminal, a função print()

Output Esperado:
Hello Python


Na segunda forma, vamos falar um pouco sobre atribuição de variáveis. Essa forma é particularmente útil para podermos alterar nosso print depois.

Para atribuir uma variável em python, basta colocar o nome da variável e o sinal de = após isso, basta atribuir o valor daquela variável.

Por exemplo:

In [3]:
cis = 'Capítulo Estudantil'
estat = 'Empresa Júnior'

Note que o nome da variável não está entre aspas, mas seu valor está. Vamos falar disso daqui a pouco, no momento é importante que vocês saibam que o nome da variável nunca é definido com aspas.

Agora é sua vez, mostre "Hello Python" no terminal por meio de uma atribuição de variáveis.

Output Esperado:
Hello Python


Note que você pode colocar qualquer nome na sua variável que só o que irá ser mostrado no terminal é seu valor.

Hello Python


# Parte 2 - Tipos de dados

O python trabalha com vários tipos de informação diferentes. Podem ser textos, números inteiros, números decimais, listas e por aí vai.

Diferente de outras linguagens, no python não é necessário declarar o tipo de uma variável para fazer ela funcionar, mas existem formas diferentes de escrever uma variável que transforma ela em cada tipo diferente.

Por exemplo:

In [6]:
texto = '1'
inteiro = 1
decimal = 1.0
lista = [1]
dicionario = {'1':1}

Observe, nesse caso, que a informação é a mesma (o número 1), ela só está cada uma com um tipo diferente. Para saber qual é o tipo de uma variável, basta usar a função type().

Sua vez, descubra o tipo das variáveis que foram declaradas ali em cima.

In [7]:
print('Output esperado:')
print(type(texto))
print(type(inteiro))
print(type(decimal))
print(type(lista))
print(type(dicionario))

Output esperado:
<class 'str'>
<class 'int'>
<class 'float'>
<class 'list'>
<class 'dict'>


Lembra quando a gente falou das aspas lá em cima? é por isso, o nome da variável não é um texto, por isso não deve ser escrito entre aspas.

Além disso, para não confundir, nomes de variáveis não podem começar com números e não podem conter pontos. 

## Parte 3 - Operadores

Agora que você sabe um pouco mais sobre os tipos de dados, nós podemos começar a fazer operações com eles.

Cada tipo de dado se comporta de forma diferente em cada operação. Por exemplo, não dá para multiplicar um texto, mas dá para controlar a quantidade de vezes que ele aparece.

In [8]:
print(texto * 3)

111


A variável 'texto' tem o valor '1', se nós multiplicarmos esse '1' por 3, ele irá aparecer 3 vezes, por isso o resultado '111'

Faz com as outras variáveis, vamos ver qual o resultado de cada uma delas:

In [16]:
texto = '1'
inteiro = 1
decimal = 1.0
lista = [1]
dicionario = {'1':1}

In [12]:
print('Output esperado para o tipo inteiro:')
print(inteiro * 3)

Output esperado para o tipo inteiro:
3


In [13]:
print('Output esperado para o tipo decimal:')
print(decimal * 3)


Output esperado para o tipo decimal:
3.0


In [17]:
print('Output esperado para o tipo lista:')
print(lista * 3)


Output esperado para o tipo lista:
[1, 1, 1]


In [18]:
print('Output esperado para o tipo dicionario:')
print(dicionario * 3)


Output esperado para o tipo dicionario:


TypeError: unsupported operand type(s) for *: 'dict' and 'int'

Note que para o tipo "dicionário" ele retorna um erro. Isso acontece porque para cada tipo existem funções e métodos diferentes. Alguns são compatíveis, outros não são. Para fazer essa operação com o dicionário vamos precisar fazer uma iteração e que já já vamos falar mais sobre

Além desse operador, podemos também somar, subtrair e por aí vai, veja no exemplo